File: KR/KR/ClientApp/src/dto/QuestionnaireInfoDto.ts
import FioDto from "./FioDto";
import PassportDetails from "./PassportDetails";
import {EmploymentType} from "./EmploymentType";
import {GoalType} from "./GoalType";
import {InformationAboutPledge} from "./InformationAboutPledge";

export default interface QuestionnaireInfoDto{
    fullName: FioDto;
    passportDetails: PassportDetails;
    age: number;
    isJudging: boolean;
    creditAmount: number;
    goal: GoalType;
    employmentType: EmploymentType;
    otherCredits: boolean;
    informationAboutPledge: InformationAboutPledge;
}

File: KR/KR/Dto/QuestionnaireInfoDto.cs
using System.ComponentModel.DataAnnotations;

namespace KR.Dto;

public class QuestionnaireInfoDto
{
    public FullNameDto FullName { get; set; }
    public PassportDetails PassportDetails { get; set; }
    [Required, Range(21, 72)]
    public int Age { get; set; }
    public bool IsJudging { get; set; }
    [Required, Range(0, 10_000_000)]
    public int CreditAmount { get; set; }
    public GoalType Goal { get; set; }
    public EmploymentType EmploymentType { get; set; }
    public bool OtherCredits { get; set; }
    public InformationAboutPledge InformationAboutPledge { get; set; }
}